pytorch_lightning/plugins/environments/torchelastic_environment.py

```diff
@@ -62,7 +62,8 @@ def main_port(self) -> int:
     def detect() -> bool:
         """Returns ``True`` if the current process was launched using the torchelastic command."""
         if _TORCH_GREATER_EQUAL_1_9_1:
-            return torch.distributed.is_torchelastic_launched()
+            # if not available (for example on MacOS), `is_torchelastic_launched` is not defined
+            return torch.distributed.is_available() and torch.distributed.is_torchelastic_launched()
```

Comment:
```suggestion
            return torch.distributed.is_available() and getattr(torch.distributed, 'is_torchelastic_launched', lambda : False)()
```
Note that for me on the mac `torch.distributed.is_available()` returns `True` which makes sense for the cpu gloo backend. So this just may not work for all the cases.

Reply:
It might be because yours has the M1. This comment says MacOS builds without it by default

https://github.com/pytorch/pytorch/blob/87148f2b5921557774a535495d7c77b20aa795b6/torch/distributed/__init__.py#L15

Then `is_torchelastic_launched` comes from this wildcard import under `is_available`:

https://github.com/pytorch/pytorch/blob/87148f2b5921557774a535495d7c77b20aa795b6/torch/distributed/__init__.py#L55